generated/osdep/meson.build

```diff
@@ -0,0 +1,68 @@
+xcrun = find_program('xcrun')
+swift_prog = find_program(run_command(xcrun, '-find', 'swift').stdout().strip())
+
+# Split the returned output based on '.'
+# The last character of the first array member is the major version.
+# We don't care about potential patch/point releases so we only fetch the
+# minor version.
+ret = run_command(swift_prog, '-version').stdout().split('.')
+swift_ver = ret[0].substring(-1) + '.' + ret[1].substring(0,1)
+
+if swift_ver.version_compare('<4.1')
+    error('A minimum Swift version of 4.1 is required!')
+elif swift_ver.version_compare('>=6.0')
+    error('Swift 6.0 and up is not supported!')
+endif
+
+# custom swift targets
+bridge = join_paths(source_root, 'osdep/macOS_swift_bridge.h')
+header = join_paths(build_root, 'osdep/macOS_swift.h')
+module = join_paths(build_root, 'osdep/macOS_swift.swiftmodule')
+target = join_paths(build_root, 'osdep/macOS_swift.o')
+
+swift_flags = ['-frontend', '-c', '-sdk', macos_sdk_path,
+               '-enable-objc-interop', '-emit-objc-header', '-parse-as-library']
+
+if get_option('debug')
+    swift_flags += '-g'
+endif
+
+if get_option('optimization') != '0'
+    swift_flags += '-O'
+endif
+
+if macos_10_11_features.allowed()
```

Comment:
hardcoding all the HAVE_* defines here depending on the usage in the swift code seems like a huge bother. tbh i don't want to add a new one here for every feature check i need in swift. in the old code all of them were just added. https://github.com/mpv-player/mpv/blob/6265724f3331e3dee8d9ec2b6639def5004a5fa2/waftools/generators/headers.py#L13-L20

Reply:
Hmm I'll see if i can try to use a foreach here to make it a little more automatic. I don't think there will be anyway around having to add a new one in the build though.

Reply:
that's pretty sad tbh.

just a general question, since i wonder if we could solve this in a better way in the future. is there a better way doing this in meson, eg in the case we remove waf and don't have to keep the compatibility anymore with the HAVE_* defines?

Reply:
Depends on what exactly you want to do.

The configuration_data object has a `.keys()` method though, if you just want to loop over all of them.